### ISSUE TITLE
Oracle: EXECUTE IMMEDIATE in a BEGIN block causes file parsing to fail

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#7043]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

The parsing of the file in the "how to reproduce" section fails when we have a the EXECUTE IMMEDIATE in the file. Commenting out that line makes it work, but the last `/` is replaced by an extra semicolon.

### Expected Behaviour

The file parsing should work. The extra `/` should not be replaced by an extra semicolon.

### Observed Behaviour

The file parsing fails.

### How to reproduce

1. Create a migration file (e.g., /workspace/migrations/example_migration.sql) as follows:

```sql
declare
    constraint_name VARCHAR2(255);
BEGIN
    SELECT constraint_name
    into CONSTRaINT_NamE
    FROM user_constraints
    wheRe table_name = 'MY_TABLE' AND constraint_type = 'C' AND search_condition_vc LIKE 'MY_CONDITION%';

    EXEcUTE Immediate 'ALTER TABLE MY_TABLE DROP CONSTRAINT ' || constraint_name;
END;
/
```

2. Configure sqlfluff as needed (see configuration section).
3. Install and run sqlfluff on that file:

```sh
pip install --no-cache-dir sqlfluff==3.4.2
sqlfluff fix /workspace/migrations/example_migration.sql
```

The response from sqlfluff is below. Note that the file parsing fails:

```
==== finding fixable violations ====
  [1 templating/parsing errors found]                                                                                                                                                                                                                        
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

4. Comment out (with `--`) or remove the line with the `EXEcUTE Immediate` and run sqlfluff again:

```sh
sqlfluff fix /workspace/migrations/example_migration.sql
```

The file is fixed:

```
==== finding fixable violations ====
== [/workspace/migrations/example_migration.sql] FAIL                                                                                                                                                                                    
L:   1 | P:   1 | CP01 | Keywords must be upper case.
                       | [capitalisation.keywords]
L:   2 | P:   1 | LT02 | Line should not be indented. [layout.indent]
L:   4 | P:   1 | LT02 | Expected indent of 2 spaces. [layout.indent]
L:   5 | P:   1 | LT02 | Expected indent of 2 spaces. [layout.indent]
L:   5 | P:   5 | CP01 | Keywords must be upper case.
                       | [capitalisation.keywords]
L:   5 | P:  10 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   6 | P:   1 | LT02 | Expected indent of 2 spaces. [layout.indent]
L:   7 | P:   1 | LT02 | Expected indent of 2 spaces. [layout.indent]
L:   7 | P:   5 | CP01 | Keywords must be upper case.
                       | [capitalisation.keywords]
L:   9 | P:   1 | LT01 | Unnecessary trailing whitespace.
                       | [layout.spacing]
L:   9 | P:   5 | LT15 | Too many consecutive blank lines.
                       | [layout.newlines]
L:  10 | P:   5 | CV06 | Statements must end with a semi-colon.
                       | [convention.terminator]
== [/workspace/migrations/example_migration.sql] FIXED
12 fixable linting violations found
```

However, the following occurs:
- We cannot parse files with `EXECUTE IMMEDIATE`, which is needed in this context.
- The final `END` has 2 semicolons, which should not be done:

We have:

```sql
END;;
```

But we should have:

```sql
END;
/
```

### Dialect

oracle

### Version

3.4.2

### Configuration

I tested it with the following .sqlfluff configuration file in /workspace:

```yaml
[sqlfluff]
templater = jinja
dialect = oracle
ignore = templating
large_file_skip_byte_limit = 0
max_line_length = 0

[sqlfluff:layout:type:alias_expression]
spacing_before = align
align_within = select_clause
spacing_after = touch

[sqlfluff:indentation]
tab_space_size = 2
indent_unit = space
indented_joins = false
indented_using_on = true
allow_implicit_indents = true
indented_on_contents = false
indented_ctes = false

[sqlfluff:rules:aliasing.table]
aliasing.table = explicit

[sqlfluff:rules:aliasing.column]
aliasing.column = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = True

[sqlfluff:rules:ambiguous.join]
fully_qualify_join_types = inner

[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = consistent

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:jinja.padding]
single_space = true

[sqlfluff:rules:layout.spacing]
no_trailing_whitespace = true
extra_whitespace = false

[sqlfluff:rules:layout.commas]
line_position = trailing

[sqlfluff:rules:layout.functions]
no_space_after_function_name = true

[sqlfluff:rules:layout.select_targets]
wildcard_policy = single

[sqlfluff:rules:layout.set_operators]
set_operator_on_new_line = ['UNION', 'UNION ALL']

[sqlfluff:rules:structure.nested_case]

[sqlfluff:rules:structure.subquery]
forbid_subquery_in = join
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by saulotoledo** [comment]
While PR #7048 successfully implements EXECUTE IMMEDIATE statement parsing, there's a separate issue with Oracle delimiter formatting that should be addressed independently:

The CV06 rule incorrectly converts Oracle's `/` statement terminator to `;;` when there's a newline between `END;` and `/`. This affects all Oracle PL/SQL blocks, not just EXECUTE IMMEDIATE statements.

Example:

```sql
-- Input:
END;
/

-- CV06 converts to:
END;;
```

Root Cause: CV06 rule treats all `statement_terminator` tokens uniformly and always replaces them with semicolons, without considering Oracle's specific `/` delimiter syntax.

This should be addressed in a separate PR focused on Oracle delimiter handling, as it requires modifications to the CV06 rule logic to properly handle Oracle-specific terminators and affects the broader Oracle dialect formatting beyond the scope of EXECUTE IMMEDIATE support.

*EDIT*: This issue has been addressed in #7054.
